google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -365,6 +380,30 @@ public Synchronicity getSynchronicity() {
     return getLogging().getWriteSynchronicity();
   }
 
+  /**
+   * Sets the metadata auto population flag.
+   *
+   * @param synchronicity {@link Synchronicity}
+   */
+  public void setAutoPopulateMetadata(Boolean value) {
```

Comment:
Do you need to make `setAutoPopulateMetadata` to be a thread safe?

Reply:
While it is possible that the same handler will be reused from multiple threads. It is less probable that multiple configuration adjustments will be run for the same handler from multiple thread in parallel.
Looking on the rest of the configuration "setters", I decided that `volatile` modifier is sufficient. WDYT?

Reply:
OK, lets keep it this way